abot/bot.py

```diff
@@ -26,15 +26,33 @@ class Abort(Exception):
 
 class Backend:
     def configure(self, **config):
+        """Feed configuration to the backend before initialization.
+
+        This method can be useful to pass usernames or passwords to the Backend
+        and store/handle them before the Bot initializates the Backend.
+        """
         raise NotImplementedError
 
     async def initialize(self):
+        """Execute one time operation which should run when the Backend starts."""
         raise NotImplementedError()
 
     async def consume(self):
+        """Execute Backend logic."""
         raise NotImplementedError()
 
     def whoami(self) -> Optional['Entity']:
+        """Return whatever entity represents the current backend.
+
+        The entity returned by this method will be use to identify the current
+        backend and match incoming Events and MessageEvents to the current
+        backend. Bear in mind that multiple backends of the same type could be
+        running at the same time. Thus, this entity should be unique for
+        each backend instance.
+
+        The idea is to leave up to the bot author to decide which entity is
+        more convenient to identify the backend.
```

Comment:
Put an example that this can be the `Entity` object that represents the bot's user

Reply:
See line 60 for example.